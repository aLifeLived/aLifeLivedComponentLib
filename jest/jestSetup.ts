/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
