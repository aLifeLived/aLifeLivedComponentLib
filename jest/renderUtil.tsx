import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
