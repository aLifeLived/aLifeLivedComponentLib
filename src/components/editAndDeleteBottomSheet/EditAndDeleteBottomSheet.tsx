import React from 'react';
import { SystemBottomSheet } from '../../themes/systemBottomSheet/SystemBottomSheet';
import { Button } from '../button/Button';
import { Box } from '../../themes/box/Box';

type EditAndDeleteBottomSheetTypes = {
  isVisible: boolean;
  onRemoveStoryPress: () => void;
  onEditStoryPress: () => void;
  onClose: () => void;
};

export const EditAndDeleteBottomSheet: React.FC<EditAndDeleteBottomSheetTypes> =
  ({ isVisible, onRemoveStoryPress, onEditStoryPress, onClose }) => {
    return (
      <SystemBottomSheet isVisible={isVisible} onClose={() => onClose()}>
        <Box width="100%" marginLeft="m" marginRight="m">
          <Button
            title="Edit story"
            onPress={() => onEditStoryPress()}
            marginBottom="s"
            backgroundColor="backgroundContrastXLow"
          />
          <Button
            title="Delete story"
            onPress={() => onRemoveStoryPress()}
            backgroundColor="backgroundError"
            color="textContrastHigh"
          />
        </Box>
      </SystemBottomSheet>
    );
  };
