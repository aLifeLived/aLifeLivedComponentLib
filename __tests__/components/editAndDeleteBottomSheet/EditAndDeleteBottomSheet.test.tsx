import React from 'react';
import { render, fireEvent } from '../../../jest/renderUtil';
import { EditAndDeleteBottomSheet } from '../../../src/components/editAndDeleteBottomSheet/EditAndDeleteBottomSheet';

const mockRemoveStoryPress = jest.fn();
const mockEditStoryPress = jest.fn();
const mockOnClose = jest.fn();

describe('components >> EditAndDeleteBottomSheet', () => {
  describe('Bottom sheet visible', () => {
    test('Should render bottom sheet correctly', () => {
      const { toJSON } = render(
        <EditAndDeleteBottomSheet
          onClose={mockOnClose}
          isVisible={true}
          onRemoveStoryPress={mockRemoveStoryPress}
          onEditStoryPress={mockEditStoryPress}
        />
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should get button titles', () => {
      const { getByText } = render(
        <EditAndDeleteBottomSheet
          onClose={mockOnClose}
          isVisible={true}
          onRemoveStoryPress={mockRemoveStoryPress}
          onEditStoryPress={mockEditStoryPress}
        />
      );
      expect(getByText('EDIT STORY')).not.toBeNull();
      expect(getByText('DELETE STORY')).not.toBeNull();
    });

    test('Should fire onRemoveStoryPress()', () => {
      const { getByText } = render(
        <EditAndDeleteBottomSheet
          onClose={mockOnClose}
          isVisible={true}
          onRemoveStoryPress={mockRemoveStoryPress}
          onEditStoryPress={mockEditStoryPress}
        />
      );

      fireEvent.press(getByText('DELETE STORY'));
      expect(mockRemoveStoryPress).toBeCalledTimes(1);
    });

    test('Should fire onEditStoryPress()', () => {
      const { getByText } = render(
        <EditAndDeleteBottomSheet
          onClose={mockOnClose}
          isVisible={true}
          onRemoveStoryPress={mockRemoveStoryPress}
          onEditStoryPress={mockEditStoryPress}
        />
      );

      fireEvent.press(getByText('EDIT STORY'));
      expect(mockEditStoryPress).toBeCalledTimes(1);
    });
  });

  describe('Bottom sheet should not be visible', () => {
    test('Should match snapshot', () => {
      const { toJSON } = render(
        <EditAndDeleteBottomSheet
          onClose={mockOnClose}
          isVisible={false}
          onRemoveStoryPress={mockRemoveStoryPress}
          onEditStoryPress={mockEditStoryPress}
        />
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
