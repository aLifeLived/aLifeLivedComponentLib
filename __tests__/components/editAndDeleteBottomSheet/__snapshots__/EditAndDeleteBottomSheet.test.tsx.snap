// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> EditAndDeleteBottomSheet Bottom sheet should not be visible Should match snapshot 1`] = `null`;

exports[`components >> EditAndDeleteBottomSheet Bottom sheet visible Should render bottom sheet correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,.5)",
        "bottom": 0,
        "display": "flex",
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "display": "flex",
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 8,
          "paddingTop": 24,
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#282828",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Edit story
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#D94E69",
          "borderRadius": 8,
          "opacity": 1,
          "padding": 16,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#282828",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Delete story
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "marginBottom": 24,
          "marginTop": 24,
          "opacity": 1,
          "padding": 16,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#282828",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;
